R/timeseries_plot.R

```diff
@@ -0,0 +1,100 @@
+# multiple stations (month)
+niger_data <- readRDS("C:/Users/lzc1n17/OneDrive - University of Southampton/PhD/IDEMS/RInstatClimatic/Niger4_day_month_year.RDS")
+niger_data <- niger_data$get_data_frame()
+niger_data_1 <- niger_data$Data
+niger_monthly_data <- niger_data$Data_by_year_month
+niger_yearly_data <- niger_data$Data_by_year
+niger_monthly_data$station = ifelse(niger_monthly_data$month_abbr %in% c("Jan", "Feb", "Mar"), "A", "B")
+
+
+# currently: ggplot, ggmiscp, viridis
+  
+
+# facets = none ----------------------
+
+ts_line_plot <- function(data, time = year, element, station = NULL, facets = "none",
```

Comment:
I think we need to decide if we are using "tidy evaluation" or not for column names i.e. `time = year` or `time = "year"`

So far I have been using strings and that would be my slight preference, but mainly because I'm more familiar with that.

For example I wouldn't know what the equivalent of `time %in% names(data)` is when doing checks if `time` is not a string. But maybe it's simple too?

Both are described here https://dplyr.tidyverse.org/articles/programming.html It doesn't seem to give much of a preference.

The difference is that you use `.data[[time]]` instead of `{{ time }}` for strings.

Do you have a preference for either?

Reply:
@volloholic and I discussed this this afternoon in a meeting - and both have a slight preference towards the "tidy evaluation" convention. For me, this is mainly because it is consistent with the tidyverse.

Perhaps after the current package deadline, this is a discussion we should reopen to decide the best move forward?

With your example, I think the best way is like this - although perhaps there's a more updated route:
```
x <- function(data, vars){
  print(vars %in% names(data))
}
x(diamonds, "color")


y <- function(data, vars){
  var_1 <- eval(vars, data)
  print(var_1 %in% names(data))
}
y(diamonds, "color")
```

Reply:
Yes, I would like to have this discussion in more detail. My preference for strings is partly because of my lack of detailed knowledge on tidy evaluation, so that's not necessarily a good reason.

Are you happy to go with strings for now while we consider tidy evaluation approaches?

Reply:
Yes absolutely happy to go with this method for now. We can discuss it after the deadline. I think it would be good to have David in that discussion as well.